improver/field_texture.py

```diff
@@ -158,11 +158,9 @@ def _calculate_ratio(self, cube, radius):
         # was zero, set ratio value to one.
 
         ratio = np.ones_like(actual_transitions.data)
-        np.divide(
-            actual_transitions.data,
-            potential_transitions.data,
-            out=ratio,
-            where=(cube.data > 0),
+        ratio[cube.data > 0] = (
+            actual_transitions.data[cube.data > 0]
+            / potential_transitions.data[cube.data > 0]
```

Comment:
This could be sped up marginally by using something like

```suggestion
            mask = cube.data > 0
            ratio[mask] = np.divide(cube.data[mask], cube2.data[mask])
```

but it's not vital, the speed up is small (21ms -> 17ms)